src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -22,6 +22,8 @@
     is_windows, is_wsl
 from azure.cli.core.cloud import get_active_cloud, set_cloud_subscription
 
+from .adal_authentication import MSIAuthenticationWrapper
```

Comment:
Another problem with `mock.patch` is for this form of import, it won't work. After importing `MSIAuthenticationWrapper`, it becomes a local reference `azure.cli.core._profile.MSIAuthenticationWrapper`. See [Where to patch](https://docs.python.org/3/library/unittest.mock.html#where-to-patch).